bundles/io/org.eclipse.smarthome.io.rest.core/OSGI-INF/configurableserviceresource.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License v1.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-v10.html
+
+-->
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="org.eclipse.smarthome.io.rest.core.service.ConfigurableServiceResource">
+   <implementation class="org.eclipse.smarthome.io.rest.core.service.ConfigurableServiceResource"/>
+   <service>
+      <provide interface="org.eclipse.smarthome.io.rest.RESTResource"/>
+      <provide interface="org.eclipse.smarthome.io.rest.core.service.ConfigurableServiceResource"/>

```

Comment:
Even a lot of the core services are directly registered without a separate interface. e.g.: ThingTypeRegistry, ChannelTypeRegistry, ItemChannelLinkRegistry, ThreadPoolManager...

If there is only one implementation we sometimes register the class directly, otherwise you always have the "Interface" "InterfaceImpl" situation. Not sure if registering classes as OSGi services is generally considered as bad practice. But I don´t see a big disadvantage.